src/NServiceBus.Persistence.NonDurable/SynchronizedStorage/NonDurableSynchronizedStorageSession.cs

```diff
@@ -0,0 +1,96 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using Extensibility;
+    using Outbox;
+    using Persistence;
+    using Transport;
+
+    class NonDurableSynchronizedStorageSession : ICompletableSynchronizedStorageSession
+    {
+        public NonDurableTransaction Transaction { get; private set; }
+
+        public void Dispose()
+        {
+            if (ownsTransaction)
```

Comment:
Why would we not set `Transaction=null` always?

Reply:
I removed the `if`.